src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3869,32 +3869,36 @@ void MarkTypeForDynamicallyAccessedMembers (ref ReflectionPatternContext reflect
 			void MarkMethodsFromReflectionCall (ref ReflectionPatternContext reflectionContext, TypeDefinition declaringType, string name, BindingFlags? bindingFlags, int? parametersCount = null)
 			{
 				bool foundMatch = false;
-				foreach (var method in declaringType.Methods) {
-					var mname = method.Name;
+				do {
+					foreach (var method in declaringType.Methods) {
+						var mname = method.Name;
 
-					if (mname != name) {
-						continue;
-					}
+						if (mname != name) {
+							continue;
+						}
 
-					if ((bindingFlags & (BindingFlags.Instance | BindingFlags.Static)) == BindingFlags.Static && !method.IsStatic)
-						continue;
+						if ((bindingFlags & (BindingFlags.Instance | BindingFlags.Static)) == BindingFlags.Static && !method.IsStatic)
+							continue;
 
-					if ((bindingFlags & (BindingFlags.Instance | BindingFlags.Static)) == BindingFlags.Instance && method.IsStatic)
-						continue;
+						if ((bindingFlags & (BindingFlags.Instance | BindingFlags.Static)) == BindingFlags.Instance && method.IsStatic)
+							continue;
 
-					if ((bindingFlags & (BindingFlags.Public | BindingFlags.NonPublic)) == BindingFlags.Public && !method.IsPublic)
-						continue;
+						if ((bindingFlags & (BindingFlags.Public | BindingFlags.NonPublic)) == BindingFlags.Public && !method.IsPublic)
+							continue;
 
-					if ((bindingFlags & (BindingFlags.Public | BindingFlags.NonPublic)) == BindingFlags.NonPublic && method.IsPublic)
-						continue;
+						if ((bindingFlags & (BindingFlags.Public | BindingFlags.NonPublic)) == BindingFlags.NonPublic && method.IsPublic)
+							continue;
 
-					if (parametersCount != null && parametersCount != method.Parameters.Count)
-						continue;
+						if (parametersCount != null && parametersCount != method.Parameters.Count)
+							continue;
 
-					foundMatch = true;
-					var methodCalling = reflectionContext.MethodCalling;
-					reflectionContext.RecordRecognizedPattern (method, () => _markStep.MarkIndirectlyCalledMethod (method, new DependencyInfo (DependencyKind.AccessedViaReflection, methodCalling)));
-				}
+						foundMatch = true;
+						var methodCalling = reflectionContext.MethodCalling;
+						reflectionContext.RecordRecognizedPattern (method, () => _markStep.MarkIndirectlyCalledMethod (method, new DependencyInfo (DependencyKind.AccessedViaReflection, methodCalling)));
+					}
+
+					declaringType = declaringType.BaseType.Resolve ();
+				} while (!foundMatch && declaringType.FullName != "System.Object");
```

Comment:
Nit: It's not necessary to check for `System.Object`, just stop at null - take a look at `MarkMethodsOnTypeHierarchy`.